### PR TITLE
Support for camel's new DSLs

### LINKED ISSUE #2294: [camel-master] Support for YAML DSL
We should create an extension for the new YAML DSL [1], an very simple extension has been created in my POC branch [2]

[1] https://github.com/apache/camel/tree/master/dsl/camel-yaml-dsl
[2] https://github.com/lburgazzoli/camel-yaml-dsl/tree/main/camel-quarkus-yaml-dsl

### LINKED ISSUE #2301: [camel-master] Support for new DSLs
relates to https://github.com/apache/camel-quarkus/issues/2294

### PR DESCRIPTION
**Pull request by lburgazzoli**
Fixes #2294 
Fixes #2301

### CODE REVIEW

**Comment by lburgazzoli** [comment]
@jamesnetherton @ppalaga the PR will certainly fail but a review would be appreciated

**Review by oscerd — APPROVED** [review]

File: extensions-core/xml-io-dsl/runtime/pom.xml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>camel-quarkus-xml-io-dsl</artifactId>
+    <name>Camel Quarkus :: XML IO DSL :: Runtime</name>
+    <description>An XML stack for parsing XML route definitions</description>
+
+    <properties>
+        <camel.quarkus.jvmSince>1.0.0</camel.quarkus.jvmSince>
+        <camel.quarkus.nativeSince>1.0.0</camel.quarkus.nativeSince>
```
Comment (jamesnetherton) [inline extensions-core/xml-io-dsl/runtime/pom.xml:35]: I guess these should be 1.8.0.

**Comment by jamesnetherton** [comment]
Question about the jOOR bits. Can they work in native mode? If not, I suppose we should add a build step to throw an exception during on native builds.

File: extensions-core/xml-io-dsl/runtime/pom.xml (hunk 1 above)
Comment (lburgazzoli) [reply extensions-core/xml-io-dsl/runtime/pom.xml:35]: In fact the functionality was there even before, so I borrowed the info from xml-io but the extensions is new, I'll bump to 1.8.0

**Comment by lburgazzoli** [comment]
> Question about the jOOR bits. Can they work in native mode? If not, I suppose we should add a build step to throw an exception during native builds.

No it won't work so I'll add make the build fail, do we have a build item for that ?

**Comment by jamesnetherton** [comment]
Here's what we did in the management extension:

https://github.com/apache/camel-quarkus/blob/master/extensions-jvm/management/deployment/src/main/java/org/apache/camel/quarkus/component/management/deployment/ManagementProcessor.java#L42-L46

**Comment by lburgazzoli** [comment]
@jamesnetherton should be a little better now

**Comment by lburgazzoli** [comment]
btw, we may have a similar situation for ServiceNow

File: extensions-core/java-joor-dsl/runtime/pom.xml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>camel-quarkus-java-joor-dsl</artifactId>
+    <name>Camel Quarkus :: Java jOOR DSL :: Runtime</name>
+    <description>Support for parsing Java route definitions at runtime</description>
+
+    <properties>
+        <camel.quarkus.jvmSince>1.8.0</camel.quarkus.jvmSince>
+        <camel.quarkus.nativeSince>1.8.0</camel.quarkus.nativeSince>
```
Comment (ppalaga) [inline extensions-core/java-joor-dsl/runtime/pom.xml:35]: Reading some code above, I had the impression that you do not want to support native. If so, plz remove camel.quarkus.nativeSince and regen the docs.
```suggestion
```

File: integration-tests/main-yaml/pom.xml [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>org.apache.camel.quarkus</groupId>
+        <artifactId>camel-quarkus-integration-tests</artifactId>
+        <version>1.8.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>camel-quarkus-integration-test-main-yaml</artifactId>
```
Comment (ppalaga) [inline integration-tests/main-yaml/pom.xml:28]: Isn't this primarily testing the new YAML DSL? If so `camel-quarkus-integration-test-yaml-dsl` would perhaps be a better name.

File: integration-tests/main-yaml/src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+## ---------------------------------------------------------------------------
+
+#
+# Main
+#
+camel.main.routes-include-pattern = classpath:routes/my-routes.yaml,classpath:routes/my-rests.yaml
```
Comment (ppalaga) [inline integration-tests/main-yaml/src/main/resources/application.properties:21]: Don't we want to coin some convention to include some pattern by default? `routes/**.yaml` maybe?

Comment (lburgazzoli) [reply integration-tests/main-yaml/src/main/resources/application.properties:21]: wildcard does not seem to work with native compilation so we either need to change the way we bundle classpath resources to native image or we have to use full paths

File: integration-tests/main-yaml/pom.xml (hunk 3 above)
Comment (lburgazzoli) [reply integration-tests/main-yaml/pom.xml:28]: yes, but we have other similar tests with the same naming pattern: should we rename all of them ?

File: integration-tests/main-yaml/src/main/resources/application.properties (hunk 4 above)
Comment (ppalaga) [reply integration-tests/main-yaml/src/main/resources/application.properties:21]: Oh, sorry, I have misread it as native resource inclusion. You are right, that wildcards neither work here nor would be easy to  implement.

File: integration-tests/main-yaml/pom.xml (hunk 3 above)
Comment (ppalaga) [reply integration-tests/main-yaml/pom.xml:28]: Sounds like a good reason not to bother with renaming :)

**Comment by ppalaga** [comment]
jOOR DSL being the first JVM-only extension in the core directory, we should double check that our tooling does what we intend:

* Extension page shouldn't say native is supported
* Do we want the extension to be listed on code.quarkus.io? - We avoided listing JVM extensions so far. code.quarkus.io does not offer any kind of JVM-only markup, IIRC.

File: extensions-core/java-joor-dsl/runtime/pom.xml (hunk 2 above)
Comment (lburgazzoli) [reply extensions-core/java-joor-dsl/runtime/pom.xml]: @ppalaga looks like the toolchain requires the field to be present as the build fails for the following reason:

```
camel.quarkus.nativeSince property must defined in extensions-core/java-joor-dsl/runtime/pom.xml
```

**Comment by lburgazzoli** [comment]
@ppalaga @jamesnetherton failures seems unrelated, should we merge and follow up and leftover ? (i.e. camel.quarkus.nativeSince for joor)

**Comment by jamesnetherton** [comment]
> @ppalaga @jamesnetherton failures seems unrelated, should we merge and follow up and leftover ? (i.e. camel.quarkus.nativeSince for joor)

Fine with me, I can look to fix up the leftovers later this week.

**Pull request merged**